app/controllers/landmarks_controller.rb

```diff
@@ -25,30 +27,30 @@ def create
       What are you stupid?!"
       render :new
     end
+  end
 
-    def edit
-      @landmark = Landmark.find(params[:id])
-    end
+  def edit
+    @landmark = Landmark.find(params[:id])
+  end
 
-    def update
-      @landmark = Landmark.find(params[:id])
-      @landmark.update(landmark_params)
-      if @landmark.save
-        flash[:notice] = "Landmark updated successfully!"
-        redirect_to landmark_path(@landmark)
-      else
-        flash[:error] = "Update unsucessful. No changes were made."
-        redirect_to edit_landmark_path(@landmark)
-      end
+  def update
+    @landmark = Landmark.find(params[:id])
+    @landmark.update(landmark_params)
```

Comment:
`update` will attempt to update and save, kind of like `create`. Put `@landmark.update` into the if instead of `update` and then `save`.